master/buildbot/process/properties.py

```diff
@@ -80,6 +82,36 @@ def master(self):
     def master(self, value):
         self._master = value
 
+    @property
+    def sourcestamps(self):
+        if self.build is not None:
+            return self.build.getAllSourceStamps()
+        return self._sourcestamps
+
+    @sourcestamps.setter
+    def sourcestamps(self, value):
+        self._sourcestamps = value
+
+    @defer.inlineCallbacks
```

Comment:
I  think we should not just pre-populate the changes as we usually already have them in reporter or schedulers.
Idea is to avoid requiring people to deal with deferred as much as we can.